### DISCUSSION TITLE
Going back to browser after paring

### DISCUSSION THREAD

**Discussion by riwa0803** [WalletConnect/walletconnect-monorepo#3851] [Developer Support] [answered]
Is there a way to automatically return to the original page after completing the pairing process with a dapp, using a deeplink from a mobile browser?

Currently, WalletConnectRouter does not work on the latest iOS.
While some wallets do have a feature to return to the originating screen on Android, this only goes back to the 'previous screen,' which is not a perfect solution.
Specifically, if you navigate away to the home screen or another app during the process, you cannot return to the original browser.

We are looking for alternative approaches that could achieve this objective, perhaps by integrating technologies other than Wallet Connect.
If not currently feasible, is there any anticipation that such a feature might be developed by third parties, or provided by Wallet Connect itself in the future?

**Comment by glitch-txs** [comment, marked as answer]
Not at the moment, as for now we suggest a UI fix approach.
https://docs.walletconnect.com/web3wallet/mobileLinking#walletconnectrouter
